dev/src/write-batch.ts

```diff
@@ -103,6 +104,39 @@ export class WriteResult {
   }
 }
 
+/** Helper type to hold a google.rpc.Status object. */
+interface RpcStatus {
```

Comment:
I think the type we actually want to return is `GoogleError` which exists in GoogleGax and matches your interface. Can you use that type instead? This allows us to use the same type in all Promise rejections.

Reply:
done inside bulkCommit()